btree/btree.c

```diff
@@ -3711,6 +3719,63 @@ int64_t btree_open_tick(struct m0_sm_op *smop)
 	}
 }
 
+/**
+ * btree_close_tree_tick function is used to traverse through different states to
+ * facilitate the working of m0_btree_open().
+ *
+ * @param smop     represents the state machine operation
+ * @return int64_t returns the next state to be executed.
+ */
+int64_t btree_close_tree_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op    *bop     = M0_AMB(bop, smop, bo_op);
+	struct m0_btree       *tree    = bop->bo_arbor;
+	struct nd             *nd_head = ndlist_tlist_head(&tree->t_desc->
+							   t_active_nds);
+	struct nd             *nd_curr;
+
+	switch (bop->bo_op.o_sm.sm_state) {
+	case P_INIT:
+	if (tree->t_desc->t_ref > 1) {
+		tree_put(tree->t_desc);
+	} else if (tree->t_desc->t_ref == 1) {
+		nd_curr = nd_head;
+		if (nd_curr->n_starttime == 0)
+			nd_curr->n_starttime = m0_time_now();
+		if (ndlist_tlist_length(&tree->t_desc->t_active_nds) > 1) {
+			/** This code is meant for debugging.
+			 *  In future, this case needs to be
+			 *  handled in a better way.
+			 */
+			if (m0_time_seconds(m0_time_now() -
+			    nd_curr->n_starttime) > 5) {
+				nd_curr->n_starttime = 0;
+				return M0_ERR(-ETIMEDOUT);
+			}
+			return P_INIT;
+		} else if (nd_curr == tree->t_desc->t_root) {
+			if (nd_curr->n_ref > 1) {
+				if (m0_time_seconds(m0_time_now() -
+				    nd_curr->n_starttime) > 5) {
+					nd_curr->n_starttime = 0;
+					return M0_ERR(-ETIMEDOUT);
+				}
+			} else {
+				nd_curr->n_starttime = 0;
+				node_put(nd_curr);
+			}
+		}
+		tree_put(tree->t_desc);
+	} else {
+		return M0_ERR(-ECANCELED);
```

Comment:
braces around single statement blocks can be optionally omitted;

Reply:
Done.

Reply:
Done.